### PR TITLE
permit indexing a case that's being created in the data registry case update repeater

### PR DESCRIPTION
**Pull request by snopoke**
## Product Description
When using the Data Registry Case Update repeater:
* Allow creating a child case of another case that's also being created in the same trasaction.

Jira: https://dimagi-dev.atlassian.net/browse/USH-1467

## Technical Summary
Check for other cases that are also being created when validating the index config.

## Feature Flag
DATA_REGISTRY_CASE_UPDATE_REPEATER

### Safety story
Only affects the one repeater class which is feature flagged. The code is well covered by tests.

### Automated test coverage
Updated

### QA Plan
None

### Migrations
NA

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/motech/repeaters/repeater_generators.py [hunk 1]
```diff
@@ -555,7 +555,7 @@ def get_case_indices(self, target_domain, target_case, configs_by_case_id):
         if not self.index_remove_case_id:
             return indices
 
-        if self.index_remove_identifier not in indices:
+        if not target_case or self.index_remove_identifier not in indices:
```
Comment (orangejenny) [inline corehq/motech/repeaters/repeater_generators.py:558]: I don't quite follow this commit. If `target_case` is None, won't this fail? https://github.com/dimagi/commcare-hq/blob/ae8b92cbadea406a2d19af9579cb7e8dcbf92ce1/corehq/motech/repeaters/repeater_generators.py#L595

Comment (snopoke) [reply corehq/motech/repeaters/repeater_generators.py:558]: Thanks, that should be inverted

Comment (snopoke) [reply corehq/motech/repeaters/repeater_generators.py]: 12f323848c8

File: corehq/motech/repeaters/tests/test_data_registry_case_update_payload_generator.py [hunk 2]
```diff
@@ -341,7 +368,7 @@ def assert_case_updates(self, expected_updates):
         """
```
[the remark this answers is not in the document]
Comment (snopoke) [reply corehq/motech/repeaters/tests/test_data_registry_case_update_payload_generator.py:368]: :+1: 4535a4f6f9d

**Review by orangejenny — APPROVED** [review]

**Pull request merged**